pandas/io/formats/style_render.py

```diff
@@ -617,9 +623,9 @@ def _format_table_styles(styles: CSSStyles) -> CSSStyles:
     return [
         item
         for sublist in [
-            [  # this is a CSSDict when TypedDict is available to avoid cast.
-                {"selector": x, "props": style["props"]}
-                for x in cast(str, style["selector"]).split(",")
+            [
+                CSSDict(selector=x, props=style["props"])
+                for x in style["selector"].split(",")
```

Comment:
I don't think you should need to create the CSSDict explicitly. mypy should check if the expression is a compatible type

```
    return [
        {"selector": selector, "props": css_dict["props"]}
        for css_dict in styles
        for selector in css_dict["selector"].split(",")
    ]
```

Reply:
just tried that and got
```
pandas/io/formats/style_render.py:624: error: List comprehension has incompatible type List[Dict[str, Union[str, List[Tuple[str, Union[str, int, float]]]]]]; expected List[CSSDict]  [misc]
```

Reply:
it maybe a mypy cache thing.

maybe better then to use for loops and append starting with css_dict : CSSDict = []

the list comp with 4 `for`s is too difficult to grok.

Reply:
hmm, seems ok

```
(pandas-dev) simon@P340:~/pandas (cssdict)$ mypy pandas
Success: no issues found in 1280 source files
(pandas-dev) simon@P340:~/pandas (cssdict)$ mypy pandas --no-incremental
Success: no issues found in 1280 source files
(pandas-dev) simon@P340:~/pandas (cssdict)$ mypy --version
mypy 0.812
(pandas-dev) simon@P340:~/pandas (cssdict)$ 
```

Reply:
> maybe better then to use for loops and append starting with css_dict : CSSDict = []

should have been css_styles : CSSStyles = []

Reply:
Seems like you're right!

Seems like
```
    ret = [
        {"selector": selector, "props": css_dict["props"]}
        for css_dict in styles
        for selector in css_dict["selector"].split(",")
    ]
    return ret
```
throws a mypy error, while
```python
    return [
        {"selector": selector, "props": css_dict["props"]}
        for css_dict in styles
        for selector in css_dict["selector"].split(",")
    ]
```
doesn't - I wasn't expecting them to be different, and I had the former pattern, sorry about that

Reply:
yep. mypy will infer the type of the intermediate variable instead of checking the return expression against the return type. could use...

ret: CSSStyles = ...
return ret